DoorGate/ViewController.swift

```diff
@@ -24,34 +26,33 @@ class ViewController: UIViewController {
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        
+
         //In Button Logic
-        inButton.rx.tap.scan(0) { (priorValue, _) in
-                return 1
-            }
-            .subscribe(onNext: { [unowned self] currentCount in
-                let count = Int(self.countLabel.text!)! + currentCount
-                self.logic(count: count)
-                })
+        inButton.rx.tap
+            .subscribe({ [unowned self] currentCount in
+                self.presenter.buttonTap(command: DoorCommand.goingIn)
+            })
             .disposed(by: disposeBag)
-       
         //Out Button Logic
-        outButton.rx.tap.scan(0) { (priorValue, _) in
-                return -1
-            }
-            .subscribe(onNext: { [unowned self] currentCount in
-                let count = Int(self.countLabel.text!)! + currentCount
-                self.logic(count: count)
+        outButton.rx.tap
+            .subscribe({ [unowned self] currentCount in
+                self.presenter.buttonTap(command: DoorCommand.goingOut)
+                })
+            .disposed(by: disposeBag)
```

Comment:
Is the disposebag disposed of for free in the in the ViewController without any action on the coders part?

Reply:
not sure what you are asking here, disposeBag is the same one declared in line 27

Reply:
I mean that we have to programatically dispose of the dispose bag in android, wasnt sure if this happens automatically.